Library/Homebrew/extend/pathname.rb

```diff
@@ -216,10 +216,10 @@ def cp_path_sub(pattern, replacement)
 
   # extended to support common double extensions
   def extname(path = to_s)
-    BOTTLE_EXTNAME_RX.match(path)
-    return $1 if $1
-    /(\.(tar|cpio|pax)\.(gz|bz2|lz|xz|Z))$/.match(path)
-    return $1 if $1
+    path =~ BOTTLE_EXTNAME_RX
+    return Regexp.last_match(1) if Regexp.last_match(1)
+    %r{(\.(dmg|tar|cpio|pax)\.(gz|bz2|lz|xz|Z|zip))$} =~ path
+    return Regexp.last_match(1) if Regexp.last_match(1)
```

Comment:
I'm also a bit curious why `zip` is included in that list imported from Cask. I wouldn't expect it to be used in this context, i.e. I've never seen `.tar.zip` or `.dmg.zip` files.